### ISSUE TITLE
Potion BUG (Health, MANA, RAGE etc.) (BB #507)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#507]
This issue was migrated from bitbucket.
**Original Reporter:** D_Jakob
**Original Date:** 26.02.2014 20:02:45 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507

<hr>

Hi guys,
I found a big bug. 
Potions don't compare max health or energy/mana/rage of caster/user, but they compare it by unit_target. 

That means if you want to drink mana potion and you are in a combat with something that doesn't have a mana, you get a message "your mana is full" and potion wasn't used.

I wonder if anyone found this bug.

Here is fix:

(Spell.ccp)

```
#!c++

@@ -4745,7 +4745,8 @@

                     if (m_spellInfo->Effect[i] == SPELL_EFFECT_HEAL)
                     {
-                        if (m_targets.getUnitTarget()->GetHealth() == m_targets.getUnitTarget()->GetMaxHealth())
+                        //if (m_targets.getUnitTarget()->GetHealth() == m_targets.getUnitTarget()->GetMaxHealth())
+                       if (p_caster->GetHealth() == p_caster->GetMaxHealth())
                         {
                             failReason = (uint8)SPELL_FAILED_ALREADY_AT_FULL_HEALTH;
                             continue;
@@ -4767,7 +4768,7 @@
                         }

                         Powers power = Powers(m_spellInfo->EffectMiscValue[i]);
-                        if (m_targets.getUnitTarget()->GetPower(power) == m_targets.getUnitTarget()->GetMaxPower(power))
+                        if (p_caster->GetPower(power) == p_caster->GetMaxPower(power))
                         {
                             failReason = (uint8)SPELL_FAILED_ALREADY_AT_FULL_POWER;
                             continue;

```

Please merge it into OregonCore, thx ;-)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** D_Jakob
**Original Date:** 26.02.2014 20:03:07 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8654233

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** D_Jakob
**Original Date:** 26.02.2014 20:04:03 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8654251

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** D_Jakob
**Original Date:** 26.02.2014 20:04:33 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8654259

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** D_Jakob
**Original Date:** 26.02.2014 20:04:45 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8654265

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** D_Jakob
**Original Date:** 26.02.2014 20:06:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8654303

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 26.02.2014 20:06:37 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8654306

<hr>

Thank you, I already noticed that issue and got fix for it, will be pushed in upcoming days.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** D_Jakob
**Original Date:** 26.02.2014 20:08:52 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8654326

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 27.02.2014 00:43:47 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8673227

<hr>

Fixed targetting of spells. Resolves #507

→ <<cset bca6285d6144>>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 02.03.2014 10:39:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/507/#comment-8866426

<hr>

Fixed targetting of spells. Resolves #507

→ <<cset a976321d18fb>>
